implementations/tutorial_coherent_vqls.py

```diff
@@ -0,0 +1,541 @@
+        A  &=  c_0 A_0 + c_1 A_1 + c_2 A_2 = \mathbb{I} + 0.2 X_0 Z_1 + 0.2 X_0, \\
+        \\
+        |b\rangle &= U_b |0 \rangle = H_0  H_1  H_2 |0\rangle,
+        \end{align}
+
+where :math:`Z_j, X_j, H_j` represent the Pauli :math:`Z`, Pauli :math:`X` and Hadamard gates applied to the qubit with index :math:`j`.
+
+This problem is computationally quite easy since a single layer of local rotations is enough to generate the
+solution state, i.e., we can use the following simple ansatz:
+
+.. math::
+        |x\rangle = V(w) |0\rangle = \Big [  R_y(w_0) \otimes  R_y(w_1) \otimes  R_y(w_2) \Big ]  H_0  H_1  H_2 |0\rangle.
+
+
+In the code presented below we solve this particular problem, by following the general scheme of the CVQLS previously discussed.
+Eventually we will compare the quantum solution with the classical one.
+
+General setup
+-------------
+This Python code requires *PennyLane* and the plotting library *matplotlib*.
+
+"""
+
+import pennylane as qml
+from pennylane import numpy as np
+import matplotlib.pyplot as plt
+
+##############################################################################
+# Setting of the main hyper-parameters of the model
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+n_qubits = 3                # Number of system qubits.
+m = 2                       # Number of ancillary qubits
+n_shots = 10 ** 6           # Number of quantum measurements.
+tot_qubits = n_qubits + m   # System + ancillary qubits.
+ancilla_idx = n_qubits      # Index of the first ancillary qubit.
+steps = 10                  # Number of optimization steps.
+eta = 0.8                   # Learning rate.
+q_delta = 0.001             # Initial spread of random quantum weights.
+rng_seed = 0                # Seed for random number generator.
+
+
+##############################################################################
+# Circuits of the quantum linear problem
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+##############################################################################
+# We need to define the unitary operations associated to the simple example
+# presented in the introduction.
+#
+# The coefficients of the linear combination are three positive numbers :math:`(1, 0.2, 0.2)`.
+# So we can embed them in the state of  :math:`m=2` ancillary qubits by adding a final zero element and
+# normalizing their sum to :math:`1`:
+
+c = np.array([1, 0.2, 0.2, 0])
+c = c / np.sum(c)
+# We also compute the square root of c
+sqrt_c = np.sqrt(c)
+
+##############################################################################
+# We need to embed the square root of the probability distribution ``c`` into the amplitudes
+# of the ancillary state. It is easy to check that one can always embed 3 positive
+# amplitudes with just three gates:
+# a local :math:`R_y` rotation, a controlled-:math:`R_y` and a controlled-NOT.
+
+
+def U_c():
+    """Unitary matrix rotating the ground state of the ancillary qubits to |sqrt(c)> = U_c |0>."""
```

Comment:
```suggestion
    """Unitary matrix rotating the ground state of the ancillary qubits 
    to |sqrt(c)> = U_c |0>."""
```
(trying to avoid the horizontal scroll bar in the built html page